symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdk.java

```diff
@@ -64,9 +64,12 @@ public SymphonyBdk(BdkConfig config) throws AuthInitializationException, AuthUna
 
   protected SymphonyBdk(BdkConfig config, ApiClientFactory apiClientFactory)
       throws AuthInitializationException, AuthUnauthorizedException {
-    this.config = config;
+    this(config, apiClientFactory, new AuthenticatorFactory(config, apiClientFactory));
+  }
 
-    final AuthenticatorFactory authenticatorFactory = new AuthenticatorFactory(config, apiClientFactory);
+  public SymphonyBdk(BdkConfig config, ApiClientFactory apiClientFactory, AuthenticatorFactory authenticatorFactory)
```

Comment:
should this constructor be public or should I create another public one initialising the ApiClientFactory like is done for the other one?

```
public SymphonyBdk(BdkConfig config, AuthenticatorFactory authenticatorFactory) throws AuthInitializationException, AuthUnauthorizedException {
    this(config, new ApiClientFactory(config), authenticatorFactory);
  }
```

Reply:
No strong opinion on that one
I would perhaps go with public SymphonyBdk(BdkConfig config, AuthenticatorFactory authenticatorFactory)
because it exposes less things?

Reply:
Or even: 
```java
public SymphonyBdk(BdkConfig config, ApiClientFactory apiClientFactory, Class<? extends AuthenticatorFactory> authenticatorFactoryClz)
      throws AuthInitializationException, AuthUnauthorizedException {

}
```
😄

Reply:
Perfect!! Thanks, I will for this one! 👍